src/library/columncache.cpp

```diff
@@ -81,7 +81,7 @@ void ColumnCache::setColumns(const QStringList& columns) {
     m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_LIBRARYTABLE_GENRE], sortNoCase);
     m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_LIBRARYTABLE_COMPOSER], sortNoCase);
     m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_LIBRARYTABLE_GROUPING], sortNoCase);
-    m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_LIBRARYTABLE_TRACKNUMBER], sortInt);
+    m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_LIBRARYTABLE_TRACKNUMBER], sortNoCase);
```

Comment:
Even though it's a string, we currently sort by integer in case they happen to parse successfully as integers -- because users will complain that "10" sorts after "1" (and as far as I can remember nobody has complained about missing "total tracks" since I've worked on Mixxx).

Now that this is a structured type, can we define a sorting on TrackNumbers that behaves in a sane way when a total number of tracks is present?